File: easy/10. Minimum Waiting Time/minimum_waiting_time.cpp
#include <algorithm>
#include <vector>

int minimumWaitingTime(std::vector<int> queries) {
  std::sort(queries.begin(), queries.end());
  int totalWaitingTime = 0;
 int q_len = queries.size();
  for (int i = 0; i < q_len; i++) {
    totalWaitingTime += (q_len - (i + 1)) * queries[i];
  }
  return totalWaitingTime;
}

File: easy/10. Minimum Waiting Time/minimum_waiting_time.go
package main

import "sort"

func MinimumWaitingTime(queries []int) int {
	sort.Ints(queries)
	minWaitingTime := 0
	q_len := len(queries)
	for i := 0; i < q_len; i++ {
		minWaitingTime += (q_len - (i + 1)) * queries[i]
	}
	return minWaitingTime
}

